src/components/button/button.component.js

```diff
@@ -41,97 +96,46 @@ const renderStyledButton = (buttonProps) => {
 
   return (
     <StyledButton
-      as={!disabled && href ? "a" : "button"} // legacy link button feature
+      as={!disabled && href ? "a" : "button"}
+      onKeyDown={href && handleLinkKeyDown}
+      draggable={false}
       buttonType={buttonType}
       disabled={disabled}
       role="button"
-      type="button"
+      type={href ? undefined : "button"}
       iconType={iconType}
       size={size}
       px={px || paddingX}
       noWrap={noWrap}
       {...tagComponent("button", buttonProps)}
-      {...styleProps}
-      ref={forwardRef}
+      {...rest}
+      ref={forwardRef || linkRef}
```

Comment:
If you provide a forwardRef I think this would break the `linkRef.current.click`.

Reply:
Sorted above by checking for `forwardRef` when space is pressed and if it exists, do `forwardRef.current.click();` instead